src/libraries/System.Net.WebSockets.Client/tests/LoopbackServer/LoopbackWebSocketServer.Http.cs

```diff
@@ -0,0 +1,185 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Net.Sockets;
+using System.Net.Test.Common;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.WebSockets.Client.Tests
+{
+    public static partial class LoopbackWebSocketServer
+    {
+        private static Task RunClientAndHttpServerAsync(
```

Comment:
Instead of this misdirection of a Func calling another Func that takes Func of a Func. I'd just make this one straight method with a bunch of `if`s, which at least could be read in a linear way.